app/models/druid_version_zip.rb

```diff
@@ -27,6 +27,19 @@ def base_key
     @base_key ||= s3_key.sub(/.zip\z/, '')
   end
 
+  # Checks to see whether a zip file already exists for this druid-version.  If it does, just touch the
+  # file to refresh atime and mtime, so the zip cache cleaning cron job doesn't see it as stale.  If it doesn't,
+  # create it.
+  # @raise [StandardError] if there's a zip file for this druid-version, but it looks too small to be complete.
+  def ensure_zip!
```

Comment:
I think `find_or_create_zip` might be a more clear name for this.

Reply:
ok, i took this suggestion.  pushing the fix exposed a pretty consistently [flappy](https://app.circleci.com/pipelines/github/sul-dlss/preservation_catalog/402/workflows/dd88079d-0b07-4da9-91e2-02712931ec19/jobs/628) test ([this one](https://github.com/sul-dlss/preservation_catalog/pull/1649/files#diff-0a92f4583841e265e22a1dbb203489bf0238d083d14fd749f73c1e0129687934R78-R87)).  per the comment i left, i wasn't wild about it, but i added a 1/10th of a second delay to that test so the `touch` would be sure to leave `atime` and `mtime` modified.  i liked that a bit more than testing that `File.touch` got called, since it tests effect more than implementation, but happy to revisit if you're more grossed out by that `sleep` call than i was (fwiw i tried 0.001 s, and still got the same test failure).